app/src/main/java/es/jvbabi/vplanplus/di/VppModule.kt

```diff
@@ -149,11 +144,28 @@ object VppModule {
             .build()
     }
 
+    @Provides
+    @Singleton
+    fun provideStringRepository(@ApplicationContext context: Context): StringRepository {
+        return StringRepositoryImpl(context)
+    }
+
+    @Provides
+    fun provideSP24NetworkRepository(): Sp24NetworkRepository {
+        return Sp24NetworkRepository()
+    }
+
+    @Provides
+    @Singleton
```

Comment:
Maybe remove Singleton since the repository can use authentication but it is shared across the whole app even after a vpp.ID was deleted resulting in multiple sources of truth

Reply:
Fixed by adapting the class everywhere necessary in 3f1cd4279ef9a00d3a9438aa934a8dcc9878b0a7.
Before every request, the auth is updated.